client/src/components/FloorDetailsForm.tsx

```diff
@@ -32,9 +32,9 @@ const FloorDetailsForm: React.FC<FloorDetailsFormProps> = ({
   floorTagState,
   submitVisibilityState,
 }): JSX.Element => {
-  if (showForm === undefined) showForm = true;
+  const [user] = useUserContext();
+  if (!showForm && user?.isAdmin) showForm = true;
```

Comment:
Maybe I would try to simplify these 2 lines because the condition is being repeated:

```typescript
if (!showform){
   showForm = user?.isAdmin
   return <></>;
}
```

Reply:
fixed